frontend/src/services/crypto.service.ts

```diff
@@ -0,0 +1,94 @@
+import 'webcrypto-shim/webcrypto-shim'
+
+const ENCRYPTION_METHOD = 'AES-GCM'
+
+async function importKey(password: string) {
+  const pwUtf8 = new TextEncoder().encode(password)
+  const pwHash = await window.crypto.subtle.digest('SHA-256', pwUtf8)
+  const key = await window.crypto.subtle.importKey(
+    'raw',
+    pwHash,
+    ENCRYPTION_METHOD,
+    false, // extractable option
+    ['encrypt', 'decrypt'] // possible actions that can be done with the key
+  )
+  return key
+}
+
+function encodeCipherToString(buffer: ArrayBuffer, iv: Uint8Array) {
```

Comment:
why is cipher arraybuffer converted to base64 but iv to hex?
how about using a library for encoding/decoding? https://www.npmjs.com/package/byte-base64